lib/evm/operation/exchange.ex

```diff
@@ -1,259 +1,17 @@
-  def swap10(_args, %{stack: _stack}) do
-    :unimplemented
-  end
-
-  @doc """
-  Exchange 11th and 12th stack items.
-
-  TODO: Implement opcode
-
-  ## Examples
-
-      iex> EVM.Operation.Exchange.swap11([], %{stack: []})
-      :unimplemented
-  """
-  @spec swap11(Operation.stack_args, Operation.vm_map) :: Operation.op_result
-  def swap11(_args, %{stack: _stack}) do
-    :unimplemented
-  end
-
   @doc """
-  Exchange 12th and 13th stack items.
-
-  TODO: Implement opcode
+  Swaps the first and last values on the stack.
 
   ## Examples
 
-      iex> EVM.Operation.Exchange.swap12([], %{stack: []})
-      :unimplemented
+      iex> EVM.Operation.Exchange.swap([1, 2, 3], %{})
+      [3, 2, 1]
+      iex> EVM.Operation.Exchange.swap([1, 2, 3, 4, 5, 6], %{})
+      [6, 2, 3, 4, 5, 1]
   """
-  @spec swap12(Operation.stack_args, Operation.vm_map) :: Operation.op_result
-  def swap12(_args, %{stack: _stack}) do
-    :unimplemented
-  end
-
-  @doc """
-  Exchange 13th and 14th stack items.
-
-  TODO: Implement opcode
-
-  ## Examples
-
-      iex> EVM.Operation.Exchange.swap13([], %{stack: []})
-      :unimplemented
-  """
-  @spec swap13(Operation.stack_args, Operation.vm_map) :: Operation.op_result
-  def swap13(_args, %{stack: _stack}) do
-    :unimplemented
-  end
-
-  @doc """
-  Exchange 14th and 15th stack items.
-
-  TODO: Implement opcode
-
-  ## Examples
-
-      iex> EVM.Operation.Exchange.swap14([], %{stack: []})
-      :unimplemented
-  """
-  @spec swap14(Operation.stack_args, Operation.vm_map) :: Operation.op_result
-  def swap14(_args, %{stack: _stack}) do
-    :unimplemented
-  end
-
-  @doc """
-  Exchange 15th and 16th stack items.
-
-  TODO: Implement opcode
-
-  ## Examples
-
-      iex> EVM.Operation.Exchange.swap15([], %{stack: []})
-      :unimplemented
-  """
-  @spec swap15(Operation.stack_args, Operation.vm_map) :: Operation.op_result
-  def swap15(_args, %{stack: _stack}) do
-    :unimplemented
-  end
-
-  @doc """
-  Exchange 16th and 17th stack items.
-
-  TODO: Implement opcode
-
-  ## Examples
-
-      iex> EVM.Operation.Exchange.swap16([], %{stack: []})
-      :unimplemented
-  """
-  @spec swap16(Operation.stack_args, Operation.vm_map) :: Operation.op_result
-  def swap16(_args, %{stack: _stack}) do
-    :unimplemented
-  end
-
-  # Helper function to push to the stack within machine_state.
-  @spec push(MachineState.t | Stack.t, EVM.val) :: Operation.op_result
-  defp push(machine_state=%MachineState{}, val) do
-    %{
-      machine_state |
-      stack: machine_state.stack
-        |> Stack.push(val)
-    }
-  end
-
-  # Helper function to just return an updated stack
-  defp push(stack, val) when is_list(stack) do
-    %{stack: Stack.push(stack, val)}
+  @spec swap(Operation.stack_args, Operation.vm_map) :: Operation.op_result
+  def swap([first | middle], _vm_map) do
+    [last | middle ] = :lists.reverse(middle)
+    [last] ++ :lists.reverse(middle) ++ [first]
```

Comment:
This last line might be faster as: `[last | :lists.reverse([first | middle])]` (if I got that right). The preference in Elixir is to avoid `++` if possible.